src/rantanplan/structures.py

```diff
@@ -11,224 +10,145 @@
+    "estrofa_unamuno": [11, 11, 7, 5],
+    "estrofa_francisco_de_la_torre": [11, 11, 11, 7],
+    "endecha_real": [7, 7, 7, 11],  # puede repetirse N veces, no sirve así
+    "lira": [7, 11, 7, 7, 11],
+    "estrofa_manriqueña": [8, 8, 4, 8, 8, 4],
+    "ovillejo_tetra": [8, 4, 8, 4, 8, 4, 8, 8, 8, 8],
+    "ovillejo_tri": [8, 3, 8, 3, 8, 3, 8, 8, 8, 8],
+    "haiku": [5, 7, 5],
+    "sonnet": 14 * [11],
+}
 
 
-def most_common_lengths(lengths_list, n=4, number_of_verses=None):
-    """Return the n most commons lengths in a list of verses lengths
-
-    :param lengths_list: List with the lengths to be checked
-    :param n: Number of most common lengths to be returned
-    :param number_of_verses: How many verses are in a given stanza
-    :return: The n most common lengths
-    """
-    most_common_lengths_list = []
-    if len(lengths_list) == number_of_verses or number_of_verses is None:
-        count = Counter(lengths_list).most_common(n)
-        for length, _ in count:
-            # `lengths` is a tuple with the length and it's count so we only get
-            # the first element of the tuple
-            most_common_lengths_list.append(length)
-    return most_common_lengths_list
-
-
-def is_seguidilla_compuesta(lengths):
+def has_minimum_length(min_length, ranges_list):
     """
-    [7, 5, 7, 5, 5, 7, 5]
-    :param lengths:
+    Checks if every range within a list of ranges contains a minimum length
+    :param min_length: Minimum length
+    :param ranges_list: List of verse length ranges
     :return:
     """
-    correct_lines = 0
-    number_of_verses = len(lengths)
-    if number_of_verses == 7:
-        for i in (0, 2, 5):
-            if 9 > lengths[i] > 5:
-                correct_lines += 1
-        for i in (1, 3, 4, 6):
-            if 7 > lengths[i] > 3:
-                correct_lines += 1
-    return correct_lines == number_of_verses
+    return all((min_length <= max(length_range) for length_range in ranges_list))
 
 
-def is_seguidilla(lengths):
+def has_maximum_length(max_length, ranges_list):
     """
-    [7, 5, 7, 5]
-    :param lengths:
-    :return:
+    Checks if every range within a list of ranges contains a maximum length
+    :param max_length: Maximum length
+    :param ranges_list: List of verse length ranges
+    :return: `True` if all verses pass the condition, `False` otherwise
     """
-    correct_lines = 0
-    number_of_verses = len(lengths)
-    if number_of_verses == 4:
-        for i in (0, 2):
-            if 9 > lengths[i] > 6:
-                correct_lines += 1
-        for i in (1, 3):
-            if 7 > lengths[i] > 4:
-                correct_lines += 1
-    return correct_lines == number_of_verses
+    return all((max_length >= min(length_range) for length_range in ranges_list))
 
 
-def is_chamberga(lengths):
+def has_fixed_length_verses(lengths_list, ranges_list, allow_fluctuation=False,
+                            fluctuation=1, repeating_structure=False):
```

Comment:
Maybe better `fluctuation_size`? Also wondering if we really need two variables or a `fluctuation_size` of 0 would effectively be the same as `allow_fluctuation=False`.
```suggestion
def has_fixed_length_verses(lengths_list, ranges_list, allow_fluctuation=False,
                            fluctuation_size=1, repeating_structure=False):
```

Reply:
Yes you're right, I'll remove allow_fluctuation

Reply:
Should `fluctuation_size` be `0` by default?

Reply:
It is now :)